hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/OnlineDsAssignmentStrategy.java

```diff
@@ -122,9 +127,26 @@ public void closeActiveActions(final List<List<Long>> targetIds) {
     @Override
     public void setAssignedDistributionSetAndTargetStatus(final JpaDistributionSet set, final List<List<Long>> targetIds,
             final String currentUser) {
-        targetIds.forEach(tIds -> targetRepository.setAssignedDistributionSetAndUpdateStatus(TargetUpdateStatus.PENDING,
-                set, System.currentTimeMillis(), currentUser, tIds));
-
+        final long now = System.currentTimeMillis();
+        targetIds.forEach(targetIdsChunk -> {
+            if (targetRepository.count(AccessController.Operation.UPDATE, targetRepository.byIdsSpec(targetIdsChunk)) != targetIdsChunk.size()) {
+                throw new InsufficientPermissionException("No update access to all targets!");
+            }
+            targetRepository.setAssignedDistributionSetAndUpdateStatus(TargetUpdateStatus.PENDING,
+                    set, now, currentUser, targetIdsChunk);
+            // TODO AC - current problem with this approach is that the caller detach the targets and seems doesn't save them
```

Comment:
Then let's stick to the old approach and perform the access control check before?